src/main.cpp

```diff
@@ -2871,6 +2883,22 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         }
     }
 
+    // History read/write is unconditional, since it is generated for entire
+    // chain regardless of the Heartwood upgrade activation height.
```

Comment:
This seems to be inconsistent with the history being per-epoch?

Reply:
Replaced with dummy condition which should be acitvated later with Heartwood upgrade code